File: javascript/rna-transcription/rna-transcription.js
const DNA_RNA_OBJECT = {
  'G' : 'C',
  'C' : 'G',
  'T' : 'A',
  'A' : 'U',
};

const corresponding = nucleotide => DNA_RNA_OBJECT[nucleotide];

const RnaTranscriber = function() {};

RnaTranscriber.prototype.toRna = function(dna) {
  const rna = dna.split('').map(corresponding);

  if(!rna.every(Boolean)) {
    throw new Error('Invalid input');
  }

  return rna.join('');
};

module.exports = RnaTranscriber;

File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription
  (:require [clojure.string :as str] ))

(def dna-to-rna-map {"G" "C" "C" "G" "T" "A" "A" "U"})

(defn trans-n [n]
  (if
    (nil? (dna-to-rna-map n))
    (throw (AssertionError. "Wrong Inptut"))
    (dna-to-rna-map n)))

(defn to-rna [string]
  (str/join ""
    (map trans-n
         (str/split string #""))))
